galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.tsq.manager.ivt/src/main/java/dev/galasa/cicsts/tsq/manager/ivt/TsqManagerIVT.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package dev.galasa.cicsts.tsq.manager.ivt;
+		
+import dev.galasa.BeforeClass;
+import dev.galasa.Test;
+import dev.galasa.core.manager.Logger;
+import org.apache.commons.logging.Log;
+
+import dev.galasa.cicsts.CicsRegion;
+import dev.galasa.cicsts.CicsTerminal;
+import dev.galasa.cicsts.CicstsManagerException;
+import dev.galasa.cicsts.ICicsRegion;
+import dev.galasa.cicsts.ICicsTerminal;
+import dev.galasa.cicsts.ITsqHandler;
+
+import dev.galasa.zos3270.FieldNotFoundException;
+import dev.galasa.zos3270.KeyboardLockedException;
+import dev.galasa.zos3270.TerminalInterruptedException;
+import dev.galasa.zos3270.spi.NetworkException;
+import dev.galasa.zos3270.TimeoutException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+	
+@Test	
+public class TsqManagerIVT {	
+	
+	@Logger
+	public Log logger;
+		
+	@CicsRegion(cicsTag="A")
+	public ICicsRegion cicsRegionA; 
+	
+	@CicsTerminal(cicsTag="A")
+	public ICicsTerminal cemtTerminalA; 
+
+	public ITsqHandler tsqA; 
```

Comment:
This should be injected with an annotation ? either getting back something that can create a TSQ, or something which IS/Represents a storage queue.

eg:
```
@CicsTemporaryStorageQueue(cicsTag="A", queueName="Q1")
public ICicsTemporaryStorageQueue queueQ1 ;

@CicsTemporaryStorageQueue(cicsTag="A", queueName="Q2", recoverable=false)
public ICicsTemporaryStorageQueue queueQ2 ;
```
or
```
@CicsTemporaryStorageQueueHandler(cicsTag="A")
public ICicsTemporaryStorageQueueHandler qHandler;
...
ICicsTemporaryStorageQueue queueQ1 = qHandler.createTemporaryStorageQueue("Q1", false) // not recoverable

ICicsTemporaryStorageQueue queueQ2 = qHandler.createTemporaryStorageQueue("Q2", false)
```

or something like that

Reply:
Not sure if an Annotated resource would be best here.  The queue doesn't really exist until data has been written to it and we don't want to do that as part of the framework as it might affect the test

Reply:
Ok @hobbit1983 . So that leaves the handler being able to do the creates.
```
@CicsTemporaryStorageQueueHandler(cicsTag="A")
public ICicsTemporaryStorageQueueHandler qHandler;
...
ICicsTemporaryStorageQueue queueQ1 = qHandler.createTemporaryStorageQueue("Q1", false) // not recoverable

ICicsTemporaryStorageQueue queueQ2 = qHandler.createTemporaryStorageQueue("Q2", false)

interface ICicsTemporaryStorageQueue {
}
```

Reply:
Updated as per the new approach to have an ITsqFactory to create a new ITsq object based on the injected queue name.